### PR TITLE
common: Set timezone (defaults to UTC)

### PR DESCRIPTION
**Pull request by zmc**
Yes, this will cause the tz to be changed on lots of our systems. I'm unsure if this could actually break things, so feedback is appreciated.

### CODE REVIEW

**Comment by zmc** [comment]
CentOS 7:
```
[ubuntu@vpm065 ~]$ # freshly provisioned
[ubuntu@vpm065 ~]$ uname -a
Linux vpm065 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
[ubuntu@vpm065 ~]$ cat /etc/redhat-release
Derived from Red Hat Enterprise Linux 7.1 (Source)
[ubuntu@vpm065 ~]$ uname -a
Linux vpm065 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
[ubuntu@vpm065 ~]$ date
Wed Jan 20 19:20:30 UTC 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 07:20:18 PM UTC  -0.391911 seconds
[ubuntu@vpm065 ~]$ # ansible -> EST
[ubuntu@vpm065 ~]$ date
Wed Jan 20 14:31:21 EST 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 02:31:07 PM EST  -0.531586 seconds
[ubuntu@vpm065 ~]$ # ansible -> UTC
[ubuntu@vpm065 ~]$ date
Wed Jan 20 19:31:47 UTC 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 07:31:34 PM UTC  -0.485914 seconds
```

**Comment by zmc** [comment]
CentOS 6:
```
[ubuntu@vpm065 ~]$ # freshly provisioned
[ubuntu@vpm065 ~]$ uname -a
Linux vpm065 2.6.32-431.23.3.el6.x86_64 #1 SMP Thu Jul 31 17:20:51 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
[ubuntu@vpm065 ~]$ date
Wed Jan 20 19:39:10 EST 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 07:39:15 PM EST  -0.438724 seconds
[ubuntu@vpm065 ~]$ # ansible -> UTC
[ubuntu@vpm065 ~]$ date
Thu Jan 21 00:40:23 UTC 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 07:40:26 PM UTC  -0.360565 seconds
[ubuntu@vpm065 ~]$ # ansible -> EST
[ubuntu@vpm065 ~]$ date
Wed Jan 20 19:40:43 EST 2016
[ubuntu@vpm065 ~]$ sudo hwclock --show
Wed 20 Jan 2016 07:40:46 PM EST  -0.031795 seconds
```

File: roles/common/tasks/yum_systems.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    dest: /etc/localtime
+    state: link
+    force: yes
+  # Default is used below to avoid breaking check mode
+  when: ansible_distribution_major_version == "6" and current_tz.stdout|default("") != timezone
+  tags:
+    - timezone
+
+- name: Set the timezone (RHEL/CentOS 7)
+  command: timedatectl set-timezone {{ timezone }}
```
Comment (dmick) [inline roles/common/tasks/yum_systems.yml]: What does this for RHEL/CentOS 6?

Comment (zmc) [inline roles/common/tasks/yum_systems.yml]: The thing immediately above, apparently

Comment (dmick) [inline roles/common/tasks/yum_systems.yml]: that would mean that everything that needs a local time is constantly reading and parsing /etc/timezone, I guess.  

I've been surprised before, but...

**Comment by zmc** [comment]
https://github.com/ceph/ceph-cm-ansible/commit/bb809c6df5d9de5da8630379e5c12cc70ff67156 sets the hardware clock on CentOS 6

File: roles/common/tasks/yum_systems.yml (hunk 1 above)
Comment (zmc) [inline roles/common/tasks/yum_systems.yml]: Turns out that wasn't setting the hardware clock; I just pushed a commit that does

File: roles/common/tasks/yum_systems.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- name: Get the current timezone (RHEL/CentOS 6)
+  shell: cut -d'"' -f2 /etc/sysconfig/clock
+  when: ansible_distribution_major_version == "6"
+  register: current_tz
+  changed_when: false
+  tags:
+    - timezone
+
+- name: Get the current timezone (RHEL/CentOS 7)
+  shell: 'timedatectl | grep Timezone | sed -e "s/.*: \(.*\) (.*/\1/"'
```
Comment (dmick) [inline roles/common/tasks/yum_systems.yml]: Still need to handle 'Time zone' I think

**Pull request merged**